From 96555b7318f37621d48834b9f9f842231c885090 Mon Sep 17 00:00:00 2001
From: Riccardo Magliocchetti <riccardo.magliocchetti@gmail.com>
Date: Tue, 9 Nov 2010 23:26:47 +0100
Subject: [PATCH 22/34] Fix cmdline parsing for alternative init system

The issue was a simple misplaced break that interrupted the cmdline
parsing at the first occurence of init=.

Should fix debian #602684 and novell #649513.
---
 bootchartd |    2 +-
 1 files changed, 1 insertions(+), 1 deletions(-)

diff --git a/bootchartd b/bootchartd
index b553368..1b032c3 100755
--- a/bootchartd
+++ b/bootchartd
@@ -165,8 +165,8 @@ if [ $$ -eq 1 ]; then
 			_init=${i#*=}
 			if test "$_init" != "/sbin/bootchartd"; then
 				init="$_init"
+				break
 			fi
-			break
 		fi
 	done
 	export PATH=$OLDPATH
-- 
1.7.3.1

